### ISSUE TITLE
Player IsMoving code

### ISSUE THREAD

**Issue by Caruychen** [RushMaverick/StuckInAWindow#3]
https://github.com/RushMaverick/StuckInAWindow/blob/3b1438fbe60e262c8b863d7caae446155697642f/src/Player.cpp#L50-L74

Is  _isMoving is really necessary? You could simplify drastically returning when (length <= 1.0f) in line 58. Here's why:

- `Player::moveToTarget` is only called by `Player::updatePosition` which checks whether `_isMoving` is true (I know it's also called in `main.cpp`, in the switch case when the player clicks, which is a redundant call).
- since you are already returning early when the length is less than 1.0f in `moveToTarget`, you don't actually need to check whether (_isMoving) is true in the parent call since it's effectively doing the same thing.
- Also, you're checking `if (player.isMoving)` in line 50 of main.cpp, to decide to call updatePosition. That's basically repeating yourself.
- Long story short, if the character is further than 1.0f from their destination, update position. If they are close enough, then don't update position.

The exception is that if you plan on adding features later than need to know whether the character is moving, then of course you can have such a boolean. But for executing the movement itself, it doesn't seem necessary.
